### PR TITLE
1478-KryptonHeaderGroup-doesnt-disappear

### PR DESCRIPTION
**Pull request by giduac**
[1478-KryptonHeaderGroup-doesnt-disappear](https://github.com/Krypton-Suite/Standard-Toolkit/issues/1478#issuecomment-2127572426)
- Throw statements removed. Even when vars are non-nullable that doesn't mean they can't be null. Inherited GetService() method can return null, and sometimes does. That's what happened in this call.
- Approach (casting and nullabilty) throughout the Action Lists slightly differed, all made equal. Applying null-forgiving and null-conditional operators.
- All cast in the constructors are done via AS.
- KListview was using the KTreeviewDesigner, now set to KlistviewDesigner and the corresponding action list
- Tested all controls on TestForm to see if they are removable now.

compile-results

### CODE REVIEW

File: Source/Krypton Components/Krypton.Navigator/Navigator/KryptonPageActionList.cs [hunk 1]
```diff
@@ -144,13 +144,13 @@ public string ToolTipBody
         /// </summary>
         public Bitmap? ImageSmall
         {
-            get => _page.ImageSmall;
+            get => _page!.ImageSmall;
 
             set
             {
-                if (_page.ImageSmall != value)
+                if (_page!.ImageSmall != value)
```
Comment (Smurf-IV) [inline Source/Krypton Components/Krypton.Navigator/Navigator/KryptonPageActionList.cs:151]: if page is null and value is null, then line 154 will still throw it's toys out of the pram !

**Review by Smurf-IV — CHANGES_REQUESTED** [review]
As the statement made, "All follow the same pattern" then the review comment will need to be checked in all areas.
Or, Page is not nullable, as it is being ignored everywhere any way !

**Comment by Wagnerp** [comment]
Please also update the changelog, so that fixes can be tracked. Thanks.

File: Source/Krypton Components/Krypton.Navigator/Navigator/KryptonPageActionList.cs (hunk 1 above)
Comment (giduac) [reply Source/Krypton Components/Krypton.Navigator/Navigator/KryptonPageActionList.cs:151]: I also noticed that , the code has been setup like that from the start and will fail in most areas if the cast from `owner.Component` to KControl should return null. Seems that this has always worked so far.

So `owner.Component` always has to be a valid reference for the action list code to run.
Putting in a check on the type cast of 'owner.Component' and throw will make use of the null-forgiving operator redundant. The control var can become non-nullable. Will process all Action Lists Like in the example below.
```cs
        private readonly KryptonPage _page;
        private readonly IComponentChangeService? _serviceComponentChange;
        private DesignerActionItemCollection _actions;
        #endregion

        #region Identity
        /// <summary>
        /// Initialize a new instance of the KryptonPageActionList class.
        /// </summary>
        /// <param name="owner">Designer that owns this action list instance.</param>
        public KryptonPageActionList(KryptonPageDesigner owner)
            : base(owner.Component)
        {
            // Remember designer and actual component instance being designed
            _page = owner.Component as KryptonPage ?? throw new NullReferenceException(GlobalStaticValues.ParameterCannotBeNull(nameof(owner.Component)));

            // Cache service used to notify when a property has changed
            _serviceComponentChange = GetService(typeof(IComponentChangeService)) as IComponentChangeService;
        }
        #endregion
        
        #region Public
        /// <summary>
        /// Gets and sets the page text.
        /// </summary>
        public string TextShort
        {
            get => _page.Text;

            set
            {
                if (!_page.Text.Equals(value))
                {
                    _serviceComponentChange?.OnComponentChanged(_page, null, _page.Text, value);
                    _page.Text = value;
                }
            }
        }

```

**Comment by giduac** [comment]
> Please also update the changelog, so that fixes can be tracked. Thanks.

Will update the changelog when this is signed off.

**Comment by Smurf-IV** [comment]
> > Please also update the changelog, so that fixes can be tracked. Thanks.
> 
> Will update the changelog when this is signed off.

Please add the update for the changelog at the same time as when the code changes are done, then it is not forgotten or split into 2 PR's for the same issue

Thanks

**Comment by giduac** [comment]
> > > Please also update the changelog, so that fixes can be tracked. Thanks.
> > 
> > 
> > Will update the changelog when this is signed off.
> 
> Please add the update for the changelog at the same time as when the code changes are done, then it is not forgotten or split into 2 PR's for the same issue
> 
> Thanks

Yes, just waiting till the changes are acccepted.
Change log added.

**Comment by Smurf-IV** [comment]
@giduac Conflicts ?

**Comment by giduac** [comment]
> @giduac Conflicts ?
The other PR interfered...
Should work now!

**Pull request merged**